File: Programmers/Lv.1/로또최고순위최저순위.py
def solution(lottos, win_nums):
    answer = [7, 7]
    for i in range(len(win_nums)):
        if win_nums[i] in lottos:
            answer[0] -= 1
            answer[1] -= 1
        if lottos[i] == 0:
            answer[0] -= 1
    if answer[0] == 7:
        answer[0] = 6
    if answer[1] == 7:
        answer[1] = 6

    return answer

# def solution(lottos, win_nums):
#     rank=[6,6,5,4,3,2,1]
#     cnt_0 = lottos.count(0)
#     ans = 0
#
#     for x in win_nums:
#         if x in lottos:
#             ans += 1
#     return rank[cnt_0 + ans],rank[ans]

File: Programmers/Lv.1.Kotlin/로또최고순위최저순위.kt
class Solution {
    fun solution(lottos: IntArray, win_nums: IntArray): IntArray {
        var answer: IntArray = intArrayOf(0, 0)
        var count = 0
        var num = 0
        
        for (i in lottos.indices) {
            if (lottos[i] == 0) {
                count++
            } else {
                if (win_nums.contains(lottos[i])) {
                    num++
                }
            }
        }
        
        answer[0] = checkNum(num + count)
        answer[1] = checkNum(num)
        
        return answer
    }
    
    fun checkNum(lotto: Int): Int {
        if (lotto == 6) {
            return 1
        } else if (lotto == 5) {
            return 2
        } else if (lotto == 4) {
            return 3
        } else if (lotto == 3) {
            return 4
        } else if (lotto == 2) {
            return 5
        } else {
            return 6
        }
    }

    // fun solution(lottos: IntArray, winNums: IntArray): IntArray {
    //     return intArrayOf(
    //             (lottos.size.plus(1)) - lottos.filter { winNums.contains(it) || it == 0 }.size,
    //             (lottos.size.plus(1)) - lottos.filter(winNums::contains).size
    //     ).map { if (it > 6) it - 1 else it }.toIntArray()
    // }
}